src/PostEventHandler.ts

```diff
@@ -3,57 +3,50 @@ import { Logger } from "@aws-lambda-powertools/logger";
+    POWERTOOLS_LOG_LEVEL = Constants.DEBUG,
+    POWERTOOLS_SERVICE_NAME = Constants.POSTEVENT_LOGGER_SVC_NAME
+} = process.env;
 
-const logger = new Logger({
-	logLevel: POWERTOOLS_LOG_LEVEL,
-	serviceName: POWERTOOLS_SERVICE_NAME,
-});
+const logger = new Logger({logLevel: POWERTOOLS_LOG_LEVEL, serviceName: POWERTOOLS_SERVICE_NAME});
 
 const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
 
 class PostEventHandler implements LambdaInterface {
-
-	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
-	async handler(event: SQSEvent, context: any): Promise<SQSBatchResponse> {
-
-		// clear PersistentLogAttributes set by any previous invocation, and add lambda context for this invocation
-		logger.setPersistentLogAttributes({});
-		logger.addContext(context);
-
-		if (event.Records.length === 1) {
-			let body;
-			const record: SQSRecord = event.Records[0];
-
-			try {
-				body = JSON.parse(record.body);
-			} catch (error) {
-				logger.error({ message:"Received invalid JSON in the SQS event record.body" });
-				return { batchItemFailures:[] };
-			}
-
-			logger.debug("Starting to process record", { event_name: body.event_name });
-
-			try {
-				await PostEventProcessor.getInstance(logger, metrics).processRequest(record.body);
-
-				logger.debug("Finished processing record from SQS");
-				return { batchItemFailures:[] };
-
-			} catch (error: any) {
-				logger.error({ message: "SQS Event could not be processed", error });
-				return { batchItemFailures:[] };
-			}
-
-		} else {
-			logger.warn({ message: "Unexpected no. of records received", numOfRecords: event.Records.length });
-			return { batchItemFailures:[] };
-		}
-	}
-
+    @metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+    async handler(event: SQSEvent, context: any): Promise<SQSBatchResponse> {
+        logger.setPersistentLogAttributes({});
+        logger.addContext(context);
+
+        if (event.Records.length !== 1) {
+            logger.warn({ message: "Unexpected no. of records received", numOfRecords: event.Records.length });
+            return { batchItemFailures: [] };
+        }
+
+        const record: SQSRecord = event.Records[0];
+        let body;
+
+        try {
+            body = JSON.parse(record.body);
+        } catch {
+            logger.error({ message: "Received invalid JSON in the SQS event record.body" });
+            return { batchItemFailures: [] };
+        }
+
+        logger.debug("Starting PostEventProcessor", { event_name: body.event_name });
+
+        try {
+            await PostEventProcessor.getInstance(logger, metrics).processRequest(record.body);
+            logger.debug("Finished processing record from SQS");
+        } catch (error: any) {
+            logger.error({ message: "SQS Event could not be processed", error });
+        }
+
+        return { batchItemFailures: [] };
```

Comment:
This is something that I've always found strange with IPR - we want to return empty batchItemFailures regardless of whether there are actually any failures or not - why is that? Does it mean that messages don't get retried even if we have a recoverable error?

Reply:
We should be doing the below as we do in GovNotify to fail it which will push it to DLQ
// explicitly set itemIdentifier to an empty string to fail the whole batch
				// see https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#services-sqs-batchfailurereporting
				batchFailures.push({ itemIdentifier: "" });
				return { batchItemFailures: batchFailures };

Reply:
Done